src/traces/histogram/calc.js

```diff
@@ -217,130 +213,158 @@ module.exports = function calc(gd, trace) {
  */
 function calcAllAutoBins(gd, trace, pa, mainData, _overlayEdgeCase) {
     var binAttr = mainData + 'bins';
-    var isOverlay = gd._fullLayout.barmode === 'overlay';
-    var i, tracei, calendar, firstManual, pos0;
+    var fullLayout = gd._fullLayout;
+    var isOverlay = fullLayout.barmode === 'overlay';
+    var i, traces, tracei, calendar, pos0, autoVals, cumulativeSpec;
+
+    var cleanBound = (pa.type === 'date') ?
+        function(v) { return (v || v === 0) ? Lib.cleanDate(v, null, pa.calendar) : null; } :
+        function(v) { return isNumeric(v) ? Number(v) : null; };
+
+    function setBound(attr, bins, newBins) {
+        if(bins[attr + 'Found']) {
+            bins[attr] = cleanBound(bins[attr]);
+            if(bins[attr] === null) bins[attr] = newBins[attr];
+        }
+        else {
+            autoVals[attr] = bins[attr] = newBins[attr];
+            Lib.nestedProperty(traces[0], binAttr + '.' + attr).set(newBins[attr]);
+        }
+    }
+
+    var binOpts = fullLayout._histogramBinOpts[trace._groupName];
 
     // all but the first trace in this group has already been marked finished
     // clear this flag, so next time we run calc we will run autobin again
     if(trace._autoBinFinished) {
         delete trace._autoBinFinished;
     }
     else {
-        // must be the first trace in the group - do the autobinning on them all
-
-        // find all grouped traces - in overlay mode each trace is independent
-        var traceGroup = isOverlay ? [trace] : getConnectedHistograms(gd, trace);
-        var autoBinnedTraces = [];
-
-        var minSize = Infinity;
-        var minStart = Infinity;
-        var maxEnd = -Infinity;
-
-        var autoBinAttr = 'autobin' + mainData;
-
-        for(i = 0; i < traceGroup.length; i++) {
-            tracei = traceGroup[i];
-
-            // stash pos0 on the trace so we don't need to duplicate this
-            // in the main body of calc
+        traces = binOpts.traces;
+        var sizeFound = binOpts.sizeFound;
+        var allPos = [];
+        autoVals = traces[0]._autoBin = {};
+        // Note: we're including `legendonly` traces here for autobin purposes,
+        // so that showing & hiding from the legend won't affect bins.
+        // But this complicates things a bit since those traces don't `calc`,
+        // hence `isFirstVisible`.
+        var isFirstVisible = true;
+        for(i = 0; i < traces.length; i++) {
```

Comment:
> Any reason for putting this here in calc as opposed to crossTraceCalc?

Makes sense, though that would necessitate essentially moving the whole `calc` step to `crossTraceCalc`, since we need these results to do the binning. Which would mean moving it to `Bar.crossTraceCalc` since histograms and bars get stacked/grouped together. Might be worthwhile at some point but for now this seemed like the more limited change.

Reply:
> Might be worthwhile at some point but for now this seemed like the more limited change.

Thanks for the info. Resolved. :ok_hand: